From 3992d7ec6fe9a0b94af7ab340e4fd16d6d5a5597 Mon Sep 17 00:00:00 2001
From: FrauH0lle <roland.goers@posteo.de>
Date: Wed, 24 Jan 2024 11:58:16 +0100
Subject: [PATCH] Capture full container command when exporting app

---
 distrobox-export | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/distrobox-export b/distrobox-export
index 77c928c21..003148d32 100755
--- a/distrobox-export
+++ b/distrobox-export
@@ -451,7 +451,7 @@ export_application() {
 		# Add closing quote
 		# If a TryExec is present, we have to fake it as it will not work
 		# through the container separation
-		sed "s|^Exec=|Exec=${container_command_prefix} |g" "${desktop_file}" |
+		sed "s|^Exec=\(.*\)|Exec=${container_command_prefix} '\1' |g" "${desktop_file}" |
 			sed "s|\(%.*\)|${extra_flags} \1|g" |
 			sed "/^TryExec=.*/d" |
 			sed "/^DBusActivatable=true/d" |
