From 182df09161fe31955b57c37483439018189e8d5e Mon Sep 17 00:00:00 2001
From: MrNerdHair <mrnerdhair@gmail.com>
Date: Mon, 25 Dec 2023 16:43:56 -0600
Subject: [PATCH] fix: binary exports should handle arguments containing spaces

---
 distrobox-export | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/distrobox-export b/distrobox-export
index 77c928c21..2c908c883 100755
--- a/distrobox-export
+++ b/distrobox-export
@@ -248,7 +248,7 @@ generate_script() {
 # name: ${container_name}
 if [ -z "\${CONTAINER_ID}" ]; then
 	exec "${DISTROBOX_ENTER_PATH:-"distrobox-enter"}" ${rootful} -n ${container_name} -- \
-		${start_shell} '${exported_bin} ${extra_flags} \$@' -- "\$@"
+		${start_shell} '${exported_bin} ${extra_flags} "\$@"' -- "\$@"
 elif [ -n "\${CONTAINER_ID}" ] && [ "\${CONTAINER_ID}" != "${container_name}" ]; then
 	exec distrobox-host-exec ${dest_path}/$(basename "${exported_bin}") ${extra_flags} "\$@"
 else
